astropy/io/registry/base.py

```diff
@@ -187,17 +188,23 @@ def delay_doc_updates(self, cls):
         because the documentation of the corresponding ``read`` and ``write``
         methods are build every time.
 
+        This context manager has no effect when running Python in optimized mode
+        (-OO CLI flag).
+
         Examples
         --------
         see for example the source code of ``astropy.table.__init__``.
         """
-        self._delayed_docs_classes.add(cls)
+        if sys.flags.optimize < 2:
```

Comment:
I normally prefer this style because it more clearly reflects that both path are branches from the same conditional, but here the new path is indeed short enough that it makes sense to write it as a guard clause.